ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -142,13 +149,21 @@ public void configure(final KsqlConfig config) {
     }
 
     ksqlConfig = config;
+    final Boolean collectMetrics = ksqlConfig.getBoolean(
+        KsqlConfig.KSQL_QUERY_PULL_METRICS_ENABLED);
+    this.pullQueryMetrics = collectMetrics
```

Comment:
pullQueryMetrics is a non final field - can we be sure that the configure method is called on the same thread that created the resource, and nothing else has seen the resource by that point, or is there some existing memory barrier being created?

If not, this is not thread safe, and another thread might see a null value of pullQueryMetrics after it has been set in wall clock time.

A solution would be to get/set pullQueryMetrics via a synchronized getter/setter or make it volatile.

If the same pattern is being used elsewhere would also need to be addressed.